### PR TITLE
Initial dump from engine

### PR DESCRIPTION
**Pull request by xmunoz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# whimsylib
+
+A game engine for text-based games, inspired by adventurelib.
+
+This is currently the game enginer behind [cronenbroguelike](https://github.com/Cronenbrogues/cronenbroguelike).
```
Comment (flosincapite) [inline README.md]: s/enginer/engine/

File: tests/common.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import unittest
+
+from whimsylib import actor
+from whimsylib.globals import G
```
Comment (flosincapite) [inline tests/common.py:4]: Now that we're doing a big refactor, I am wondering what import style is best, particularly for this global object.

```
from whimsylib.globals import G
from whimsylib.globals import G as _G
```

Both have been used throughout this and the cronenbroguelike code base. @xmunoz and @abjose, any opinions?

File: tests/whimsylib/bag_test.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        backpack.add(pamphlet)
+        self.assertIs(backpack, pamphlet.holder)
+
+    @patch("whimsylib.actor.say.insayne")
+    def test_add_aliases_message_for_player(self, mock_say):
+        pamphlet = item.Item("sycophantic propaganda pamphlet")
+        G.player.inventory.add(pamphlet)
+        mock_say.assert_called_once_with(f"You acquire {pamphlet.name}.")
+
+    # TODO: Might wants messages for NPCs someday, you know.
```
Comment (flosincapite) [inline tests/whimsylib/bag_test.py]: You can remove this TODO; it's unnecessary.

**Comment by xmunoz** [comment]
There are two failing tests:
```
FAILED tests/whimsylib/say_test.py::InsayneTest::test_no_alteration_when_insanity_low - AssertionError: '' != 'hereisnodeathinalltheworld.'
FAILED tests/whimsylib/say_test.py::InsayneTest::test_output_reconstructable - AssertionError: 'When Gregor Samsa awoke one morning from u[74 chars]min.' != 'W'
```
@flosincapite can you take a look?

**Comment by flosincapite** [comment]
> There are two failing tests:
> 
> ```
> FAILED tests/whimsylib/say_test.py::InsayneTest::test_no_alteration_when_insanity_low - AssertionError: '' != 'hereisnodeathinalltheworld.'
> FAILED tests/whimsylib/say_test.py::InsayneTest::test_output_reconstructable - AssertionError: 'When Gregor Samsa awoke one morning from u[74 chars]min.' != 'W'
> ```
> 
> @flosincapite can you take a look?

Yep!

File: tests/whimsylib/say_test.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    for oc in original:
+        while True:
+            try:
+                next_aug = next(aug_gen)
+            except StopIteration:
+                break
+
+            if oc == next_aug:
+                reconstructed.append(next_aug)
```
Comment (flosincapite) [inline tests/whimsylib/say_test.py:45]:
```suggestion
                reconstructed.append(next_aug)
                break
```

Comment (flosincapite) [reply tests/whimsylib/say_test.py:45]: Somehow, we lost that break statement, but tests aren't failing in cronenbroguelike.

This is what happens when you write a body horror game: it ends up haunted.

File: tests/whimsylib/actor_test.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+from unittest.mock import call
```
Comment (flosincapite) [inline tests/whimsylib/actor_test.py:1]: Out of curiosity, why do we no longer have a `unit/` subdirectory in `tests/`?

File: tests/common.py (hunk 2 above)
Comment (xmunoz) [reply tests/common.py:4]: I prefer the first one. The second is misleading.

File: tests/whimsylib/actor_test.py (hunk 5 above)
Comment (xmunoz) [reply tests/whimsylib/actor_test.py:1]: Nope! We can change the dir structure once we have other kinds of tests, but for now let's keep it simple.

File: tests/common.py (hunk 2 above)
Comment (xmunoz) [reply tests/common.py:4]: Alright just pushed a change that removes `as _G` everywhere.

**Comment by xmunoz** [comment]
> > There are two failing tests:
> > ```
> > FAILED tests/whimsylib/say_test.py::InsayneTest::test_no_alteration_when_insanity_low - AssertionError: '' != 'hereisnodeathinalltheworld.'
> > FAILED tests/whimsylib/say_test.py::InsayneTest::test_output_reconstructable - AssertionError: 'When Gregor Samsa awoke one morning from u[74 chars]min.' != 'W'
> > ```
> > 
> > 
> > @flosincapite can you take a look?
> 
> Yep!

Fixed by adding the `break` back in.

**Comment by flosincapite** [comment]
> > > There are two failing tests:
> > > ```
> > > FAILED tests/whimsylib/say_test.py::InsayneTest::test_no_alteration_when_insanity_low - AssertionError: '' != 'hereisnodeathinalltheworld.'
> > > FAILED tests/whimsylib/say_test.py::InsayneTest::test_output_reconstructable - AssertionError: 'When Gregor Samsa awoke one morning from u[74 chars]min.' != 'W'
> > > ```
> > > 
> > > 
> > > @flosincapite can you take a look?
> > 
> > 
> > Yep!
> 
> Fixed by adding the `break` back in.

:fried_egg:

**Review by flosincapite — APPROVED** [review]

**Review by flosincapite — APPROVED** [review]

**Pull request merged**